### ISSUE TITLE
Source YouTube Analytics: implement monetary.readonly scope for the single google cloud workspace 

### ISSUE THREAD

**Issue by igrankova** [airbytehq/airbyte#19017]
## Tell us about the problem you're trying to solve
To verify the scope 'https://www.googleapis.com/auth/yt-analytics-monetary.readonly' we need to implement it on the prod Airbyte Cloud. Google requires the scope implementation for the production-ready applications only _(see emails with Subject: OAuth Verification Request for Project prod-ab-cloud-proj (id: prod-ab-cloud-proj) on the integration-test@airbyte.io)_.

## Describe the solution you’d like
In order not to affect existing users, implement this scope request for **oauth-testing** workspace only. 
LastPass creds name: **airbyte.io**

After the verification this scope should be added to the YouTube Analytics source for all workspaces.

**Comment by bazarnov** [comment]
@girarda @evantahler 

Could you please assist with enabling the `source-youtube-analytics` for the single workspace on Cloud, in order to pass the verification from the part of google? I think we did something similar in the past but for Shopify.

Once the scope is approved, I'll revert the changes back to normal.

**Comment by igrankova** [comment]
Cloud workspace count criteria - 3

**Comment by bazarnov** [comment]
> @igrankova @bazarnov  Can you clarify what you mean affected? Would these users lose their syncs? If so, we'll need to proactively reach out to the customers and inform them.

I think this could be real, never did the changes like this, @girarda would you be able to advice here?

**Comment by girarda** [comment]
@bazarnov @YowanR we can't do the same hack we did for shopify because the youtube connector is already enabled and used by customers.

In an ideal world, we would be able to publish an updated version of the connector with the new scope only for their test-workspace, but 
1. workspace-scoped connector definitions are not enabled on cloud yet ([slack thread on timeline](https://airbytehq-team.slack.com/archives/C03A8CSANPQ/p1667835084976929))
2. the oauth flow is part of the platform, not the connector so all workspaces use the same flow

Possible workarounds:
1. Update the scopes we request (this will require proper QA on the dev instances to ensure we don't break existing workflows
2. Create a new source identical to source-youtube-analytics, but requesting the monetary.readonly scope. Google should be able to certify the connector if they only have access to this new source instead of the existing one. The new source can be deleted when we apply the update to the real one after the cert. (very hacky)

maybe @airbytehq/connector-operations team has more ideas on how we can roll this out safely

**Comment by bazarnov** [comment]
@girarda thank you for jump in, I agree we can go with the option #2, and revert this afterwards.

**Comment by bazarnov** [comment]
Daily Updates:
- steps `1 - 4` are done, in: https://github.com/airbytehq/airbyte/pull/19223

Continue tomorrow with `5 - 8` steps.

@girarda 

As for: 
> Create a new source identical to source-youtube-analytics, but requesting the monetary.readonly scope. Google should be able to certify the connector if they only have access to this new source instead of the existing one. The new source can be deleted when we apply the update to the real one after the cert. (very hacky)

How can we limit the new source to the single workspace with current `webapp` structure? do we need to do this in cloud repo or OSS? Thank you, in advance.

**Comment by timroes** [comment]
> How can we limit the new source to the single workspace with current webapp structure? do we need to do this in cloud repo or OSS? Thank you, in advance.

Not the best solution, but atm we're filtering out sources like that in the FE. You can check this file: https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/core/domain/connector/constants.ts#L32-L34

**Comment by bazarnov** [comment]
@timroes 
Got it, this is ok for this one, I'll make another line for this demo case `source-youtube-analytics-business` and will revert it after google's approval.

@igrankova What workspace will we use for the demo?

**Comment by bazarnov** [comment]
@igrankova the demo-case connector is available for https://cloud.airbyte.io/workspaces/d705a766-e9e3-4689-85cb-52143422317d

**Comment by erica-airbyte** [comment]
@bazarnov can this issue be closed?

**Comment by bazarnov** [comment]
We can move it to On hold, but not close it. We wait for Google approvals around this scope for Youtube.

**Comment by igrankova** [comment]
@bazarnov, @YowanR the scope is verified!

**Comment by YowanR** [comment]
Best news ever! 🙏 
@YuliiaNahoha @oustynova Can you try to prioritize this towards the end of the month, if we have the bandwidth?

**Comment by oustynova** [comment]
@YowanR sure, will prioritize!

**Comment by bazarnov** [comment]
The 2 PRs are created and are already in Internal Review:
1) https://github.com/airbytehq/airbyte-platform-internal/pull/5222
2) https://github.com/airbytehq/airbyte/pull/24148

They should be merged in the indicated order as they were created.
Waiting for the approvals.

**Comment by bazarnov** [comment]
The `source-youtube-analytics` was modified with the new `business` scope in OAuth Flow part, the Customers should be able to use their business accounts along with the regular `Youtube Analytics` source. Closing this now.
